concepts/Common/Security Considerations/30 Export Vulnerabilities/05 CSV Injection.md

```diff
@@ -0,0 +1,191 @@
+If you use DataGrid or PivotGrid Export functionality, you should consider a threat of a CSV Injection Attack (also called formula injection attack). It involves the injection of harmful characters that act as a formula and are executed within a computer network. Cell values that begin with the **=, +, -** and **@** characters can initiate an injection attack.
+
+When you enter the malicious code into CSV files (in the form of a formula), the formula can be executed. Once executed, the malicious formula can manipulate/delete user data or provide unauthorized access to data/internal resources.
+
+You can encode the CSV files to prevent execution of harmful code in the exported files. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.
```

Comment:
> You can encode **the** CSV files to prevent execution of harmful code in the exported files.

"CSV files" are plural here and require no article unless we have a very good reason to emphasize them.

> You can encode **the CSV files** to prevent execution of harmful code in **the exported files**.

We speak of the same thing, then why should we repeat ourselves?

I can also note that no code can be executed once we encode all executable fragments in an exported CSV file.

```suggestion
You can encode CSV files to prevent execution of potentially harmful code in them. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.
```